src/dataExplorer/components/Results.tsx

```diff
@@ -144,58 +124,157 @@ const Results: FC = () => {
+  const {view, setView /* setViewOptions*/} = useContext(ResultsViewContext)
+
+  const updateChildResults = useCallback(
+    update => {
+      setView({
+        ...view,
+        properties: {
+          ...view.properties,
+          ...update,
+        },
+      })
+    },
+    [setStatus, setResult]
+  )
+
+  // TODO: make component, for QxBuilder-specific graph subquery options
+  const subQueryOptions = null
+
+  return (
+    <>
+      {subQueryOptions}
+      {/* <SubQueryOptions update={setViewOptions} /> */}
+      <ViewOptions
+        properties={view.properties}
+        results={result.parsed}
+        update={updateChildResults}
+      />
+    </>
+  )
+}
+
+const GraphHeader: FC = () => {
+  const {view, setView} = useContext(ResultsViewContext)
+  const {result} = useContext(ResultsContext)
+  const {result: subQueryResult} = useContext(ChildResultsContext)
+  const {launch} = useContext(SidebarContext)
+
+  const launcher = () => {
+    launch(<WrappedOptions />)
   }
 
-  const dataExists = res && Object.entries(res).length
   const updateType = viewType => {
     setView({
       state: 'graph',
       properties: SUPPORTED_VISUALIZATIONS[viewType].initial,
     })
   }
 
-  const launcher = () => {
-    launch(<WrappedOptions />)
+  const dataExists = !!result?.parsed
+  const subqueryReturnsData = !!subQueryResult?.parsed
+  let titleText = 'Configure Visualization'
+  if (!dataExists) {
+    titleText = 'No data to visualize yet'
+  }
+  if (!subqueryReturnsData) {
+    titleText = 'Graph customization options returned no data'
+  }
+
+  return (
+    <>
+      <ViewTypeDropdown
+        viewType={view.properties.type}
+        onUpdateType={updateType}
+      />
+      <Button
+        text="Customize"
+        icon={IconFont.CogSolid_New}
+        onClick={launcher}
+        status={dataExists ? ComponentStatus.Default : ComponentStatus.Disabled}
+        color={ComponentColor.Default}
+        titleText={titleText}
+        className="de-config-visualization-button"
+      />
+    </>
+  )
+}
+
+const Results: FC = () => {
+  const [search, setSearch] = useState('')
+  const {status} = useContext(ResultsContext)
+  const {view, setView} = useContext(ResultsViewContext)
+
+  let resultView
+
+  if (status === RemoteDataState.NotStarted) {
+    resultView = <EmptyResults />
+  } else {
+    if (view.state === 'table') {
+      resultView = <TableResults search={search} />
+    } else {
+      resultView = <GraphResults />
+    }
   }
 
-  const tableHeader =
+  const updateViewState = state => {
+    if (state === 'graph') {
+      setView({
+        state: 'graph',
+        properties: SUPPORTED_VISUALIZATIONS['xy'].initial,
+      })
+    } else {
+      setView({
+        state: 'table',
+        properties: SUPPORTED_VISUALIZATIONS['simple-table'].initial,
+      })
+    }
+  }
+
+  const Header =
     view.state === 'table' ? (
       <>
         <div style={{width: '300px'}}>
```

Comment:
For a minor re-rendering optimization, this can be pulled out into a variable defined outside the `GraphHeader` component. As it stands, this will cause unnecessary re-renders, because JavaScript treats the object literal `{width: '300px'}` as not the same object when doing comparisons on re-renders because each re-render creates a new object which doesn't have the same memory address as the object created on the previous re-render.

This can be solved by storing this in a variable outside of the component it's in and used in place of the object literal:

```ts
const headerStyle = {width: '300px'}
const GraphHeader: FC = () => {
...

      <div style={headerStyle}
```

The only reason I even bring up such a minor optimization is that this code is on the render path of graphs, which can have a tendency to get janky and chuggy.

Reply:
I learned something. Yes!